### ISSUE TITLE
fps display in the progress test should better be instant ?

### ISSUE THREAD

**Issue by 44vince44** [DJATOM/x265-aMod#9]
Hello @DJATOM !

Currently the progress text of x265 shows important information, all calculated properly, but it seems that fps is calculated as average from the beginning:

fps = (nb of frames processed since beginning) / (time elapsed since beginning).

This gives a  idea of the global behavior in the process. 

However, fps is a bench-marking information. It has to reflect current status of processing speed.
Let's say, there is suddenly another process launched on the computer, then the fps should almost immediately drop. But it doesn't because the fps is a long term average here. So, I suggest:

fps = (nb of frames processed on a sliding window, or in a buffer) / (time used to process those frames)

That of course, is in my humble opinion.
And thanks always for the great work!

**Comment by DJATOM** [comment]
I personally think that average fps is more meaningful then instant. But sure, I'll consider it and probably make an option to calculate fps with different strategies.

**Comment by 44vince44** [comment]
I totally agree that all fps methods have uses and pros and cons indeed. Thanks for anything you would do 😄 !

**Comment by Dendraspis** [comment]
Just an idea:
The current fps would not be very useful, because it might jump a lot. But instead an additional 5min average could be nice/interesting. 🤔

**Comment by 44vince44** [comment]
@Dendraspis agreed, "pure" instant values have no use at all ! 
Also @DJATOM suggested that the "range" used to calculate the average can be set by the user.

However, the `Total Size` and` Time Remaining` should always use the global average values (exactly as they are currently calculated) otherwise they would be unreliable.

**Comment by 44vince44** [comment]
I have added an element of information here: https://github.com/Patman86/x265-Mod-by-Patman/issues/4

Hopefully, "someone" will take a look at it 😁

**Comment by DJATOM** [comment]
Unfortunately I don't have much time to look into that feature in few weeks or so, as I've got a load of work 😞
